application/src/main/java/org/togetherjava/tjbot/features/help/HelpThreadCreatedListener.java

```diff
@@ -56,20 +58,19 @@ public HelpThreadCreatedListener(HelpSystemHelper helper) {
     }
 
     @Override
-    public void onChannelCreate(ChannelCreateEvent createEvent) {
-        if (!createEvent.getChannelType().isThread()) {
-            return;
-        }
-        ThreadChannel threadChannel = createEvent.getChannel().asThreadChannel();
-
-        if (wasThreadAlreadyHandled(threadChannel.getIdLong())) {
-            return;
-        }
-
-        if (!helper.isHelpForumName(threadChannel.getParentChannel().getName())) {
-            return;
+    public void onMessageReceived(@NotNull MessageReceivedEvent event) {
+        if (event.isFromThread()) {
+            IThreadContainerUnion parentChannel =
```

Comment:
you can make this variable of type `Channel` instead, `IThreadContainerUnion` looks weird in the first place and is not needed as you only use `Channel#getName` from it

Reply:
sure thing